src/TextInput.js

```diff
@@ -1,167 +1,156 @@
-import React, { Component } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
-
-    if (props.password) {
-      this.id = `password${idgen()}`;
+const TextInput = ({
+  value,
+  children,
+  s,
+  m,
+  l,
+  xl,
+  disabled,
+  noLayout,
+  placeholder,
+  icon,
+  label,
+  inputClassName,
+  success,
+  error,
+  password,
+  email,
+  validate,
+  defaultValue,
+  type,
+  ...props
+}) => {
+  const newId = id => {
+    id = id || idgen();
+
+    if (password) {
+      id = `password${idgen()}`;
     }
 
-    if (props.email) {
-      this.id = `email${idgen()}`;
+    if (email) {
+      id = `email${idgen()}`;
     }
-  }
+    return id;
+  };
 
-  componentDidMount() {
+  const [newValue, setNewValue] = useState(value);
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
     if (typeof M !== undefined) {
```

Comment:
This statement will always return `true` because `typeof` will always return a string.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof
> The **typeof** operator returns a string indicating the type of the unevaluated operand.

The right way would be `if (typeof M !== 'undefined') {`, but since this will change the behavior of this component would be better to check it with the repository owners.

Reply:
We should just drop this check altogether.
Let it throw an error if `M` is undefined

Reply:
agreed, this check was wrong since the beginning since it just made the component fail silently